rocAL/rocAL_pybind/amd/rocal/plugin/generic.py

```diff
@@ -38,50 +38,63 @@ def __init__(self, pipeline, tensor_layout = types.NCHW, reverse_channels = Fals
         self.batch_size = pipeline._batch_size
         if self.loader._name is None:
             self.loader._name = self.loader._reader
-        self.labels_size = ((self.bs*self.loader._numOfClasses) if (self.loader._oneHotEncoding == True) else self.bs)
-        self.out = self.dimensions = self.dtype = None
-        self.len = b.getRemainingImages(self.loader._handle)//self.bs # iteration length
+        self.labels_size = ((self.batch_size * self.loader._num_classes) if self.loader._one_hot_encoding else self.batch_size)
+        self.output_list = self.dimensions = self.dtype = None
```

Comment:
Not used here but it will be used if we plan to add copyTensor changes. Since all iterators have the same generic template so multiplier and offset are added in every iterator and the values are passed from the pipeline

Reply:
@SundarRajan28 Has the copyOutTensor been incorporated ?

If yes, check if we need the multiplier and offset

Reply:
copyOutTensor has been implemented in C++ but pybind has not been added since all trainings runs fine with copy_data. The multiplier and offset is present here just in case we need to add the support for copyTensor in any trainings